src/main/java/de/tum/in/www1/artemis/web/rest/FileUploadExerciseResource.java

```diff
@@ -254,6 +257,12 @@ else if (!authCheckService.isAtLeastTeachingAssistantForExercise(fileUploadExerc
         List<GradingCriterion> gradingCriteria = gradingCriterionRepository.findByExerciseIdWithEagerGradingCriteria(exerciseId);
         fileUploadExercise.setGradingCriteria(gradingCriteria);
 
+        List<Feedback> feedbackList = feedbackRepository.findFeedbackByStructuredGradingInstructionId(gradingCriteria);
+
+        if (!feedbackList.isEmpty()) {
+            fileUploadExercise.setGradingInstructionFeedbackUsed(true);
+        }
```

Comment:
Do you have a service that you can move this since the same code appears for each exercise type?

Reply:
same thing for this part. I am defining new service methods for the second part, I'll use service for this work also.